lib/galaxy/webapps/galaxy/controllers/user.py

```diff
@@ -78,24 +78,21 @@ def openid_auth( self, trans, **kwd ):
         if not trans.app.config.enable_openid:
             return trans.show_error_message( 'OpenID authentication is not enabled in this instance of Galaxy' )
         message = 'Unspecified failure authenticating via OpenID'
-        openid_url = kwd.get( 'openid_url', '' )
-        openid_provider = kwd.get( 'openid_provider', '' )
-        if not openid_provider or openid_url:
-            openid_provider = trans.app.openid_providers.NO_PROVIDER_ID  # empty fields cause validation errors
-        redirect = kwd.get( 'redirect', '' ).strip()
         auto_associate = util.string_as_bool( kwd.get( 'auto_associate', False ) )
         use_panels = util.string_as_bool( kwd.get( 'use_panels', False ) )
-        action = 'login'
         consumer = trans.app.openid_manager.get_consumer( trans )
+        openid_url = kwd.get( 'openid_url', '' )
+        openid_provider = kwd.get('openid_provider', '')
         if openid_url:
             openid_provider_obj = trans.app.openid_providers.new_provider_from_identifier( openid_url )
+        elif openid_provider:
+            openid_provider_obj = trans.app.openid_providers.get(openid_provider)
         else:
-            openid_provider_obj = trans.app.openid_providers.get( openid_provider )
-        if not openid_url and openid_provider == trans.app.openid_providers.NO_PROVIDER_ID:
             message = 'An OpenID provider was not specified'
-        elif openid_provider_obj:
-            if not redirect:
-                redirect = ' '
+        redirect = kwd.get('redirect', '').strip()
+        if not redirect:
+            redirect = ' '
+        if openid_provider_obj:
```

Comment:
Sorry I'm late on this!
I think that if both `openid_url` and `openid_provider` are empty, `openid_provider_obj` would not be defined, resulting in a `NameError` exception at this line.